### PR TITLE
Reorganized content.

### PR DESCRIPTION
**Pull request by achatterjee-grafana**
Brokedown one big troubleshooting topic into individual topics.

### CODE REVIEW

File: docs/tempo/website/troubleshooting/_index.md [hunk 1]
```diff
@@ -4,152 +4,11 @@ weight: 470
...
-- [Problem 3. Getting error message ‘Too many jobs in the queue’](#problem-3-getting-error-message-too-many-jobs-in-the-queue)
-- [Problem 4. Distributor is not accepting traces](#problem-4-distributor-is-not-accepting-traces)
+The topics in this section helps with day zero operational issues that may come up when getting started with Tempo. They walk you through debugging each part of the ingestion and query pipeline to drill down and diagnose issues.
 
-## Problem 1. I am unable to see any of my traces in Tempo
-**Potential causes**
-- There could be issues in ingestion of the data into Tempo, that is, spans are either not being sent correctly to Tempo or they are not getting sampled.
-- There could be issues querying for traces that have been received by Tempo.
+- [I am unable to see any of my traces in Tempo](unable-to-see-trace)
+- [Some of my traces are missing in Tempo](unable-to-see-trace)
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/_index.md:11]: nit: this should link to the missing-traces doc

File: docs/tempo/website/troubleshooting/_index.md [hunk 2]
```diff
@@ -4,152 +4,11 @@ weight: 470
...
+- [I am getting error message ‘Too many jobs in the queue’](too-many-jobs-in-queue)
+- [I have reached the maximum trace limit](max-trace-limit-reached)
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/_index.md:13]: The user would not know this, can we replace with `Seeing <... max-live-traces..> log line in Tempo` OR `Per tenant limit overrides`

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Check whether the application spans are actually reaching Tempo. The following metrics help determine this
+- `tempo_distributor_spans_received_total`
+- `tempo_ingester_traces_created_total`
+
+The value of both metrics should be greater than `0` within a few minutes of the application spinning up.
+You can check both metrics using -
+- The metrics page exposed from Tempo at `http://<tempo-address>:<tempo-http-port>/metrics`, or
+- In Prometheus, if it is being used to scrape metrics
+ 
+## Issue 1 - `tempo_distributor_spans_received_total` is 0
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:23]:
```suggestion
### Case 1 - `tempo_distributor_spans_received_total` is 0
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+---
+
+
+# I am unable to see any of my traces in Tempo
+
+**Potential causes**
+- There could be issues in ingestion of the data into Tempo, that is, spans are either not being sent correctly to Tempo or they are not getting sampled.
+- There could be issues querying for traces that have been received by Tempo.
+
+## Diagnosing and fixing ingestion issues
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:13]:
```suggestion
## Section 1: Diagnosing and fixing ingestion issues
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+#### Fixing sampling issues
+- These issues can be tricky to determine because most SDKs use a probabilistic sampler by default. This may lead to just one in a 1000 records being picked up.
+- Check the sampling configuration of the tracer being initialized in the application and make sure it has a high sampling rate.
+- Some clients also provide metrics on the number of spans reported from the application, for example `jaeger_tracer_reporter_spans_total`. Check the value of that metric if available and make sure it is greater than zero.
+- Another way to diagnose this problem would be to generate lots and lots of traces to see if some records make their way to Tempo.
+
+#### Fixing incorrect endpoint issue
+- If the application is also running inside docker, make sure the application is sending traces to the correct endpoint (`tempo:<receiver-port>`).
+
+## Issue 2 - `tempo_ingester_traces_created_total` is 0
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:46]:
```suggestion
## Case 2 - `tempo_ingester_traces_created_total` is 0
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+If the value of `tempo_ingester_traces_created_total` is 0, the possible reason is -
+- Network issues between distributors and ingesters.
+
+This can also be confirmed by checking the metric `tempo_request_duration_seconds_count{route='/tempopb.Pusher/Push'}` exposed from the ingester which indicates that it is receiving ingestion requests from the distributor.
+
+### Solution
+- Check logs of distributors for a message like `msg="pusher failed to consume trace data" err="DoBatch: IngesterCount <= 0"`.
+  This is likely because no ingester is joining the gossip ring, make sure the same gossip ring address is supplied to the distributors and ingesters.
+
+## Diagnosing and fixing issues with querying traces
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:56]:
```suggestion
## Section 2: Diagnosing and fixing issues with querying traces
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+ 
+## Issue 1 - `tempo_distributor_spans_received_total` is 0
+If the value of `tempo_distributor_spans_received_total` is 0, possible reasons are:
+- Use of incorrect protocol/port combination while initializing the tracer in the application.
+- Tracing records not getting picked up to send to Tempo by the internal sampler.
+- Application is running inside docker and sending traces to an incorrect endpoint.
+
+Receiver specific traffic information can also be obtained using `tempo_receiver_accepted_spans` which has a label for the receiver (protocol used for ingestion. Ex: `jaeger-thrift`).
+
+### Solutions
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:31]:
```suggestion
#### Solutions
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+##### Fixing protocol/port problems
+- Find out which communication protocol is being used by the application to emit traces. This is unique to every client SDK. For instance: Jaeger Golang Client uses `Thrift Compact over UDP` by default.
+- Check the list of supported protocols and their ports and ensure that the correct combination is being used. You will find the list of supported protocols and ports here: https://grafana.com/docs/tempo/latest/getting-started/#step-1-spin-up-tempo-backend
+
+#### Fixing sampling issues
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:37]:
```suggestion
##### Fixing sampling issues
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+- These issues can be tricky to determine because most SDKs use a probabilistic sampler by default. This may lead to just one in a 1000 records being picked up.
+- Check the sampling configuration of the tracer being initialized in the application and make sure it has a high sampling rate.
+- Some clients also provide metrics on the number of spans reported from the application, for example `jaeger_tracer_reporter_spans_total`. Check the value of that metric if available and make sure it is greater than zero.
+- Another way to diagnose this problem would be to generate lots and lots of traces to see if some records make their way to Tempo.
+
+#### Fixing incorrect endpoint issue
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:43]:
```suggestion
##### Fixing incorrect endpoint issue
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+- If the application is also running inside docker, make sure the application is sending traces to the correct endpoint (`tempo:<receiver-port>`).
+
+## Issue 2 - `tempo_ingester_traces_created_total` is 0
+If the value of `tempo_ingester_traces_created_total` is 0, the possible reason is -
+- Network issues between distributors and ingesters.
+
+This can also be confirmed by checking the metric `tempo_request_duration_seconds_count{route='/tempopb.Pusher/Push'}` exposed from the ingester which indicates that it is receiving ingestion requests from the distributor.
+
+### Solution
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:52]:
```suggestion
#### Solution
```

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+- `tenant-id not found`
+
+Possible reasons for the above errors are:
+- Tempo Querier not connected to Tempo Query Frontend. Check the value of the metric `cortex_query_frontend_connected_clients` exposed by the Query Frontend.
+  It should be > 0, which indicates that Queriers are connected to the Query Frontend.
+- Grafana Tempo Datasource not configured to pass tenant-id in Authorization header (only applicable to multi-tenant deployments).
+- Not connected to Tempo Querier correctly
+- Insufficient permissions
+
+### Solution
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:79]:
```suggestion
#### Solutions
```

File: docs/tempo/website/troubleshooting/_index.md [hunk 12]
```diff
@@ -4,152 +4,11 @@ weight: 470
...
+The topics in this section helps with day zero operational issues that may come up when getting started with Tempo. They walk you through debugging each part of the ingestion and query pipeline to drill down and diagnose issues.
 
-## Problem 1. I am unable to see any of my traces in Tempo
-**Potential causes**
-- There could be issues in ingestion of the data into Tempo, that is, spans are either not being sent correctly to Tempo or they are not getting sampled.
-- There could be issues querying for traces that have been received by Tempo.
+- [I am unable to see any of my traces in Tempo](unable-to-see-trace)
+- [Some of my traces are missing in Tempo](missing-trace)
+- [I am getting error message ‘Too many jobs in the queue’](too-many-jobs-in-queue)
+- [I have reached the per tenant limit overrides](max-trace-limit-reached)
```
Comment (annanay25) [inline docs/tempo/website/troubleshooting/_index.md:13]:
```suggestion
- [I have reached the max trace limit](max-trace-limit-reached)
```

Sorry to send you on the wrong path, the original message made sense here. Re-adding with this suggestion.

**Review by annanay25 — APPROVED** [review]
Thanks for this, I think the individual pages are a lot better. Let's merge after these small changes.

File: docs/tempo/website/troubleshooting/_index.md (hunk 1 above)
Comment (achatterjee-grafana) [reply docs/tempo/website/troubleshooting/_index.md]: Done

**Pull request merged**